backend/sozisel/lib/sozisel_web/schema/resolvers/participant_resolvers.ex

```diff
@@ -0,0 +1,109 @@
+defmodule SoziselWeb.Schema.Resolvers.ParticipantResolvers do
+  alias Sozisel.Repo
+  alias SoziselWeb.Schema.Helpers
+  alias SoziselWeb.Schema.Subscriptions.Topics
+
+  alias Sozisel.Model.{
+    Sessions,
+    Events,
+    LaunchedEvents,
+    Participants,
+    Quizzes,
+    EventResults,
+    Utils
+  }
+
+  alias Events.Event
+  alias LaunchedEvents.LaunchedEvent
+  alias Sessions.Session
+  alias Participants.Participant
+  alias Quizzes.{QuizResult, ParticipantAnswer, TrackNode}
+
+  require Logger
+
+  # password is optional so we can't always pattern match on all arguments
+  def join_session(
+        _parent,
+        %{input: %{email: email, full_name: full_name, session_id: session_id} = input},
+        _ctx
+      ) do
+    entry_password = Map.get(input, :entry_password)
+
+    with %Session{entry_password: ^entry_password} <- Repo.get_by(Session, id: session_id),
+         {:ok, %Participant{} = participant} <-
+           Participants.create_participant(%{
+             session_id: session_id,
+             email: email,
+             full_name: full_name
+           }) do
+      {:ok, %{token: participant.token}}
+    else
+      {:error, reason} ->
+        Logger.error("Failed to create a participant with reason: #{inspect(reason)}")
+
+        {:error, "failed to create a participant"}
+
+      _ ->
+        {:error, "unauthorized"}
+    end
+  end
+
+  def submit_quiz_results(
+        _parent,
+        %{
+          input: %{
+            launched_event_id: launched_event_id,
+            participant_answers: participant_answers
+          }
+        },
+        ctx
+      ) do
+    with %LaunchedEvent{} = launched_event <-
+           LaunchedEvents.get_launched_event(launched_event_id),
+         %LaunchedEvent{session: %Session{}, event: %Event{}} = launched_event <-
+           Repo.preload(launched_event, [:session, :event]),
+         %{context: %{session: session_ctx, participant: participant}} <- ctx,
+         true <- launched_event.session.id == session_ctx.id do
+      event_questions = Map.get(launched_event.event.event_data, :quiz_questions)
+
+      participant_answers =
+        Enum.map(event_questions, fn event_question ->
+          answer_on_question =
+            participant_answers |> Enum.find(&(&1.question_id == event_question.id))
```

Comment:
One more styling comment, when you do the pipe operations try to keep them one under another in a column. Same when you do `Enum.map` it is prettier to do
```
some_fields
|> Enum.map(fn ...

...)
```
Not in all cases of course but here you have a lot of code and it is more readable.